android/java/org/chromium/chrome/browser/util/TabUtils.java

```diff
@@ -10,23 +10,139 @@
 import android.app.Activity;
 import android.content.Context;
 import android.content.Intent;
+import android.os.Build;
 import android.view.ContextThemeWrapper;
 import android.view.MenuItem;
 import android.view.View;
 import android.widget.FrameLayout;
 import android.widget.PopupMenu;
 
+import androidx.core.content.ContextCompat;
+import androidx.core.view.MenuCompat;
+
 import org.chromium.base.ApplicationStatus;
+import org.chromium.base.supplier.ObservableSupplier;
 import org.chromium.chrome.R;
 import org.chromium.chrome.browser.ChromeTabbedActivity;
 import org.chromium.chrome.browser.app.BraveActivity;
 import org.chromium.chrome.browser.app.ChromeActivity;
+import org.chromium.chrome.browser.bookmarks.BookmarkModel;
+import org.chromium.chrome.browser.bookmarks.BookmarkUtils;
 import org.chromium.chrome.browser.crypto_wallet.util.WalletConstants;
 import org.chromium.chrome.browser.night_mode.GlobalNightModeStateProviderHolder;
+import org.chromium.chrome.browser.tab.Tab;
 import org.chromium.chrome.browser.tab.TabLaunchType;
+import org.chromium.chrome.browser.toolbar.LocationBarModel;
+import org.chromium.components.bookmarks.BookmarkId;
 import org.chromium.content_public.browser.LoadUrlParams;
 
+import java.lang.reflect.Field;
+import java.lang.reflect.Method;
+
 public class TabUtils {
+    public static void showBookmarkTabPopupMenu(Context context, View view,
+            ObservableSupplier<BookmarkModel> bookmarkModelSupplier,
+            LocationBarModel locationBarModel) {
+        Context wrapper = new ContextThemeWrapper(context, R.style.BookmarkTabPopupMenu);
+
+        PopupMenu popup = new PopupMenu(wrapper, view);
+        popup.getMenuInflater().inflate(R.menu.bookmark_tab_menu, popup.getMenu());
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.Q) {
+            popup.setForceShowIcon(true);
+        } else {
+            try {
+                Field[] fields = popup.getClass().getDeclaredFields();
+                for (Field field : fields) {
+                    if ("mPopup".equals(field.getName())) {
+                        field.setAccessible(true);
+                        Object menuPopupHelper = field.get(popup);
+                        Class<?> classPopupHelper =
+                                Class.forName(menuPopupHelper.getClass().getName());
+                        Method setForceIcons =
+                                classPopupHelper.getMethod("setForceShowIcon", boolean.class);
+                        setForceIcons.invoke(menuPopupHelper, true);
+                        break;
+                    }
+                }
+            } catch (Exception e) {
+                e.printStackTrace();
+            }
+        }
+
+        Tab currentTab;
+        BookmarkModel bridge;
+        if (locationBarModel != null) {
```

Comment:
```suggestion
        if (locationBarModel != null) {
```
currentTab = locationBarModel != null ? locationBarModel.getTab() : null;